tests/test_connection.py

```diff
@@ -271,10 +271,23 @@ def test_client_error_response(self):
 
         exception = context_manager.exception
         eq_(request_id, exception.request_id)
-        eq_(error_message, str(exception))
+        eq_("Errors found processing batch update ([{u'index': 0, u'error': {u'status': u'error', u'message': u'Email address  is invalid'}}])",
+            str(exception))
         eq_(invalid_emails, exception.error_data['invalidEmails'])
         eq_(failure_messages, exception.error_data['failureMessages'])
 
+    def test_str_client_exception_when_no_failure_messages(self):
```

Comment:
Minor point but the name of this function could be `test_str_client_exception_without_failure_messages`